Create /etc/sysconfig if directory does not exist

TomasKohout opened the pull request:
<!--  Thanks for sending a pull request!

Please be aware that we're following the Kubernetes guidelines of contributing
to this project. This means that we have to use this mandatory template for all
of our pull requests.

Please also make sure you've read and understood our contributing guidelines
(https://github.com/cri-o/cri-o/blob/main/CONTRIBUTING.md) as well as ensuring
that all your commits are signed with `git commit -s`.

Here are some additional tips for you:

- If this is your first time, please read our contributor guidelines:
  https://git.k8s.io/community/contributors/guide#your-first-contribution and
  developer guide
  https://git.k8s.io/community/contributors/devel/development.md#development-guide
- Please label this pull request according to what type of issue you are
  addressing, especially if this is a release targeted pull request. For
  reference on required PR/issue labels, read here:
  https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
- If you want *faster* PR reviews, read how:
  https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
- If the PR is unfinished, see how to mark it:
  https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?

<!--
Uncomment only one `/kind <>` line, hit enter to put that in a new line, and
remove leading whitespace from that line:
-->

<!--
/kind api-change

/kind ci
/kind cleanup
/kind dependency-change
/kind deprecation
/kind design
/kind documentation
/kind failing-test
/kind feature
/kind flake
/kind other
-->
/kind bug
#### What this PR does / why we need it:
Since we are moving from docker to cri-o, I've tried to create correct configuration for Ubuntu OS. When I've specified correct path to irqbalance_config_file which is by the documentation located at /etc/default/irqbalance for Ubuntu, crio failed to start because crio supposed that /etc/sysconfig directory exists. 
#### Which issue(s) this PR fixes:

<!--
Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->

<!--
Fixes #
or
None
-->

#### Special notes for your reviewer:
I'd be glad if someone could point me in the right direction about some unit test for this. :slightly_smiling_face: 
#### Does this PR introduce a user-facing change?
No.
<!--
If no, just write `None` in the release-note block below. If yes, a release note
is required: Enter your extended release note in the block below. If the PR
requires additional action from users switching to the new release, include the
string "action required".

For more information on release notes see:
https://git.k8s.io/community/contributors/guide/release-notes.md
-->

```release-note
none
```

haircommander commented on internal/runtimehandlerhooks/high_performance_hooks.go:
```diff
@@ -360,6 +362,14 @@ func RestoreIrqBalanceConfig(irqBalanceConfigFile, irqBannedCPUConfigFile, irqSm
 	}
 	if !fileExists(irqBannedCPUConfigFile) {
+		// check if folder exists, if not create /etc/sysconfig
+		if !fileExists(EtcSysconfig) {
```
I don't think we can use fileexists, as it checks the path is not a directory (which we expect it to be). Maybe just do
```
if _, err := os.Stat(filename); os.IsNotExist(err) {
...
}
```

TomasKohout replied:
You're right. I oversaw the info.IsDir() method call.

haircommander commented:
thanks for the contribution and for trying out CRI-O @TomasKohout ! unit tests are in `internal/runtimehandlerhooks/high_performance_hooks_test.go` if you want to try to add one. I left one nit
/ok-to-test

TomasKohout commented:
Hi @haircommander ,

I've added the unit test and changed a bit the logic of getting the directory path (it was ugly for the first time anyway :smile: ). Would you be so kind and check if everything is ok? 

Thank you. :slightly_smiling_face:

TomasKohout commented:
/retest

TomSweeneyRedHat commented:
Code changes LGTM, I'm not sure what's going on with the tests, but there is an awful lot of red unhappiness there.

TomasKohout commented:
Hi @TomSweeneyRedHat ,

yep, I'm also not sure what's wrong. Is Jenkins or some dependent tool ok?

haircommander commented:
it's defeinitely unrelated to this PR, no to worry

haircommander commented:
/retest

haircommander commented:
heya @TomasKohout can you rebase your PR to fix the CI issues?

haircommander commented:
can you drop the merge commit and do a rebase instead? I recommend:

```
git fetch upstream
git rebase upstream/main
git push -f origin main
```

TomasKohout commented:
It's done. :)

haircommander commented:
/retitle Create /etc/sysconfig if directory does not exist

finnicky CI :)

haircommander commented:
/release-note none

haircommander commented:
/retest-required

TomasKohout commented:
/retest

haircommander commented:
/lgtm

thanks for the patience on this one @TomasKohout !

haircommander commented:
/approve

TomasKohout commented:
Hi @haircommander , NP. :-) 

Should I do a rebase or why do you think the tests are failing? I wan unable to figure it out from provided logs.
